build-logic/publication/src/main/kotlin/convention.publish-base.gradle.kts

```diff
@@ -9,6 +11,16 @@ version = providers.gradleProperty("projectVersion")
     .forUseAtConfigurationTime()
     .get()
 
+tasks.withType(Jar::class.java) {
+    manifest {
+        attributes(
+            mapOf(
+                Attributes.Name.IMPLEMENTATION_VERSION.toString() to project.version
+            )
+        )
+    }
+}
```

Comment:
Added a release version to all JAR's.

Possible use cases:
- Metadata for troubleshooting. E.g. then we see a JAR file as an intermediate artifact in Gradle cache.
- A client wants to use the new implementation in advance and it's backward compatible
- We want to add a version-specific logic in infrastructure. Now we can only add a comment `// TODO: do something after 2021.X release`. We could check it programmatically.

Reply:
An alternative option is to pass the infra version as a parameter. But it's error-prone and leads to duplication. Clients need to know it while configuring build scripts.

Reply:
i like attribute version and agree with use cases,  nice!